### DISCUSSION TITLE
No available formula or cask with the name "gcc". MacOS version 11.2.3 .I need the latest g++ to use pbds.

### DISCUSSION THREAD

**Discussion by guptajiakshat** [Homebrew/discussions#1048] [Getting started]
% brew config
HOMEBREW_VERSION: 3.0.5-57-g0cf2d30
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0cf2d3039d8529ac750b1c2431a4c25617b61d54
Last commit: 44 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/brew
Core tap HEAD: 0cf2d3039d8529ac750b1c2431a4c25617b61d54
Core tap last commit: 44 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A

**Comment by SMillerDev** [comment]
Run `brew doctor` and follow the instructions. You messed up the homebrew/homebrew-core git repo.

**Reply by guptajiakshat** [reply]
yes after that ?

**Reply by guptajiakshat** [reply]
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://github.com/Homebrew/brew

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

**Reply by SMillerDev** [reply]
Here is the instruction you need to follow

> With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
`git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core`

**Reply by guptajiakshat** [reply]
what do i write in terminal ?

**Reply by gromgit** [reply]
```
git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core
```

**Reply by guptajiakshat** [reply]
% brew install gcc
Updating Homebrew...
/usr/local/Homebrew/Library/Homebrew/brew.sh: line 100: /usr/local/bin/brew: No such file or directory
/usr/local/Homebrew/Library/Homebrew/brew.sh: line 218: /usr/local/bin/brew: No such file or directory
/usr/local/Homebrew/Library/Homebrew/brew.sh: line 218: exec: /usr/local/bin/brew: cannot execute: No such file or directory

**Reply by gromgit** [reply]
What's the output of the following:
```bash
git -C "/usr/local/Homebrew" remote -v
git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote -v
```

**Reply by guptajiakshat** [reply]
origin	https://github.com/Homebrew/homebrew-core (fetch)
origin	https://github.com/Homebrew/homebrew-core (push)


origin	https://github.com/Homebrew/homebrew-core (fetch)
origin	https://github.com/Homebrew/homebrew-core (push)

**Reply by gromgit** [reply]
Looks like an old problem: Your core tap wasn't properly initialized as a Git repository, so every Git operation on it was actually modifying the core software repo instead.

To confirm this, what's the output of:
```
git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" rev-parse --show-toplevel
```

First fix the core software repo:
```
git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew
git -C "/usr/local/Homebrew" fetch --force origin
git -C "/usr/local/Homebrew" fetch --force --tags origin
git -C "/usr/local/Homebrew" reset --hard origin/master
```
Check that the core software is correctly installed by running `brew --repo homebrew/core`. This command **must print ONLY** `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core`. If you see anything different, or anything extra, show us what you see.

Once you've gotten past that stage, now you can fix the core tap:
```
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
```
If this command throws any errors, show us what you see.

If there are no errors, show us the output of `brew config` and `brew doctor`.

**Reply by guptajiakshat** [reply]
/usr/local/Homebrew

% brew config
HOMEBREW_VERSION: 3.0.5-59-gd1fd7e3
ORIGIN: https://github.com/Homebrew/brew
HEAD: d1fd7e31e4366948145b357de77b33f100b54889
Last commit: 34 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 1a08e7e164be79e2197380a346758bdc29442956
Core tap last commit: 17 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A

% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:

**Reply by gromgit** [reply]
OK, now try to `brew install` whatever you need.

Also:
> ```zsh
> % git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" rev-parse --show-toplevel
> /usr/local/Homebrew
> ```

That confirms my original guess. The output of the above command should've been `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core`.

**Reply by guptajiakshat** [reply]
yes its running thankyou very much
